dao/src/main/java/se/fortnox/reactivewizard/db/ObservableStatementFactory.java

```diff
@@ -0,0 +1,114 @@
+
+import static java.lang.String.format;
+import static rx.Observable.error;
+
+public class ObservableStatementFactory {
+
+    public static final  int    QUERY_LOG_THRESHOLD_MS = 5000;
+    public static final  int    RECORD_BUFFER_SIZE     = 100000;
+    private static final Logger LOG                    = LoggerFactory.getLogger("Dao");
+    private final ConnectionProvider         connectionProvider;
+    private final DbStatementFactory         statementFactory;
+    private final PagingOutput               pagingOutput;
+    private final Scheduler                  scheduler;
+    private final Metrics                    metrics;
+    private final Function<Object[], String> paramSerializer;
+
+    public ObservableStatementFactory(@Nullable ConnectionProvider connectionProvider,
+        DbStatementFactory statementFactory,
+        PagingOutput pagingOutput,
+        Scheduler scheduler,
+        Function<Object[], String> paramSerializer,
+        Metrics metrics
+    ) {
+        this.connectionProvider = connectionProvider;
+        this.statementFactory = statementFactory;
+        this.pagingOutput = pagingOutput;
+        this.scheduler = scheduler;
+        this.metrics = metrics;
+        this.paramSerializer = paramSerializer;
+    }
+
+    private static void closeSilently(Connection connection) {
+        try {
+            connection.close();
+        } catch (SQLException sqlException) {
+            LOG.warn("Failed to close connection", sqlException);
+        }
+    }
+
+    public Observable<Object> create(Object[] args) {
+        AtomicReference<TransactionStatement> transactionHolder = new AtomicReference<>();
+
+        Observable<Object> result = Observable.create(subscription -> {
+
+            Statement dbStatement = statementFactory.create(args, subscription);
+
+            TransactionStatement transaction = transactionHolder.get();
+            if (transaction != null) {
+                transaction.setConnectionProvider(connectionProvider);
+                transaction.executeStatement(dbStatement);
+            } else {
+                executeStatement(dbStatement);
+            }
+        });
+
+        if (DebugUtil.IS_DEBUG || LOG.isDebugEnabled()) {
+            Exception queryFailure = new RuntimeException("Query failed");
+            result = result.onErrorResumeNext(thrown -> {
+                queryFailure.initCause(thrown);
+                return error(queryFailure);
+            });
+        }
+
+        result = pagingOutput.apply(result, args);
+        result = result.onBackpressureBuffer(RECORD_BUFFER_SIZE);
+        result = metrics.measure(result, time -> logSlowQuery(time, args));
+        result = LoggingContext.transfer(result);
+        result = result.subscribeOn(scheduler);
+
+        return new DaoObservable<>(result, transactionHolder);
+    }
+
+    private void logSlowQuery(long time, Object[] args) {
+        if (time > QUERY_LOG_THRESHOLD_MS) {
```

Comment:
This has been changed in the internal repo.

Reply:
We'll do another PR to sync changes that occured during migration afterwards, it'll be easier :)